File: Problem1/Answ.c
#include <stdio.h>
#define MAX 10000


int Answ(int arr[], int targ) {
    int length = sizeof(arr) / sizeof(int);
    for (int i = 0; i < length; i++)
    {   
        int s[MAX] = {0};
        int temp = targ - arr[i];
        if (s[temp] == 1) {
            return 0;
        }
        s[arr[i]] = 1;
    }
    return 1;
    
}
int main() {
    int arr[] = {10, 15, 3, 7};
    int targ = 13;
    int answer = Answ(arr, targ);
    if (answer == 1) {
        printf("True\n");
    } else {
        printf("False\n");
    }
    return 0;
}

File: src/Problem1/Java/Answ.java
import java.util.ArrayList;

public class Answ {
    private int array[], targ;
    private int holder[] = new int[5];
    public Answ(int array[], int targ ) {
        this.array = array;
        this.targ = targ;
    }

    public boolean answer() {
        boolean check = true;
        ArrayList tempArray = new ArrayList<Integer>();
        int temp;
        for (int i = 0; i < array.length; i++) {
            temp = targ - array[i];
            if (tempArray.contains(temp)) {
                return check;
            }
            tempArray.add(array[i]);
        }
        return !check;
    }

    public static void main(String[] args) {
        Answ prb = new Answ(new int[] {10, 15, 3, 7}, 13);
        System.out.println(prb.answer());
    }
}